src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -451,22 +452,30 @@ public FilterElement build()
         {
             public void serialize(FilterElement operation, DataOutputPlus out, int version) throws IOException
             {
-                out.writeBoolean(operation.isDisjunction);
                 out.writeUnsignedVInt(operation.expressions.size());
                 for (Expression expr : operation.expressions)
                     Expression.serializer.serialize(expr, out, version);
+
```

Comment:
It is not clear for me that we are allowed to downgrade version this way.
Perhaps only a certain class of filters can be downgraded (those with `isDisjunction == false`? those which are childless?)

Reply:
I have added an assertion to handle this situation.